odoo/tools/translate.py

```diff
@@ -443,8 +443,16 @@ def _get_lang(self, frame):
                     lang = env['res.users'].context_get()['lang']
         return lang
 
-    def __call__(self, source):
-        return self._get_translation(source)
+    def __call__(self, source, *args, **kw):
+        translation = self._get_translation(source)
+        if args or kw:
+            try:
+                return translation % (args or kw)
+            except (TypeError, ValueError, KeyError):
+                bad = translation
+                translation = source % (args or kw)
+                _logger.error('Bad translation %r for %r', bad, source)
```

Comment:
How about
```suggestion
                _logger.error('Bad translation %r for %r', translation, source)
                return source % (args or kw)
```
Alternatively, if we want a single point of return, replace 450 by `translation = translation % (args or kw)`. And either way, lifting the error to right after the `except` clause seems like a good idea (the `bad` local seems wholly unnecessary).

Reply:
I though about it but I wanted to log only if the issue was because of a bad translation or a bad source. If it fails in English as well, you shouldn't get a "Bar translation" error, no ?

Reply:
My suggestion doesn't change that? It just moves things around a bit to avoid the unnecessary locale. If it fails in english as well then the error is not caught by either version, it just blows up.

Reply:
```
2020-06-02 09:24:31,122 38822 ERROR mc odoo.tools.translate: Bad translation 'Text avec %s marqueur' for string 'Text with %s placeholder' 
Traceback (most recent call last):
  File "/home/mat/odoo/odoo/odoo/tools/translate.py", line 450, in __call__
    return translation % (args or kw)
TypeError: not all arguments converted during string formatting
```
The `Bad translation 'Text avec %s marqueur' for string 'Text with %s placeholder'` should not be there as the translation is actually correct, it's a programming error, not a translation issue.

Reply:
Well yes, all I'm saying is I don't think my suggestion changes that, it's already there in the original no?